core-bundle/src/Controller/BackendPreviewSwitchController.php

```diff
@@ -88,12 +90,28 @@ public function __invoke(Request $request): Response
         return new Response('', Response::HTTP_BAD_REQUEST);
     }
 
+    public function setBackendConfig(array $attributes, string $badgeTitle): void
+    {
+        $this->backendAttributes = $attributes;
+        $this->backendBadgeTitle = $badgeTitle;
+    }
+
     private function renderToolbar(): string
     {
         $canSwitchUser = $this->security->isGranted('ROLE_ALLOWED_TO_SWITCH_MEMBER');
         $frontendUsername = $this->tokenChecker->getFrontendUsername();
         $showUnpublished = $this->tokenChecker->isPreviewMode();
 
+        $attributes = '';
+
+        if (!empty($this->backendAttributes)) {
+            $attributes = ' '.implode(' ', array_map(
+                static fn ($v, $k) => sprintf('data-%s="%s"', $k, $v),
+                $this->backendAttributes,
+                array_keys($this->backendAttributes)
+            ));
+        }
+
```

Comment:
HTML composition should be done in the template, not in the controller. IMHO this part can be dropped and the `backendAttributes` directly passed to the template. 

Once we introduce a `html_attr` function or something alike (compare to [here](http://craftsnippets.com/articles/using-attr-function-to-render-html-attributes-in-craft-cms) or [here](https://www.drupal.org/docs/8/theming-drupal-8/using-attributes-in-templates)) this allows a simple use inside templates. Plus you can semantically add/remove things.

Reply:
I refactored the template variable for the attributes to an array. Once the new twig function is available it can be refactored in the template!